enterprise/coderd/httpmw/provisionerdaemon_test.go

```diff
@@ -0,0 +1,224 @@
+package httpmw_test
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/v2/coderd/httpmw"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/enterprise/coderd/coderdenttest"
+	"github.com/coder/coder/v2/enterprise/coderd/license"
+	"github.com/coder/coder/v2/testutil"
+)
+
+func TestExtractProvisionerDaemonAuthenticated(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name                    string
+		opts                    httpmw.ExtractProvisionerAuthConfig
+		expectedStatusCode      int
+		expectedResponseMessage string
+		provisionerKey          string
+		provisionerPSK          string
+	}{
+		{
+			name: "NoKeyProvided_Optional",
+			opts: httpmw.ExtractProvisionerAuthConfig{
+				DB:       nil,
+				Optional: true,
+			},
+			expectedStatusCode:      http.StatusOK,
+			expectedResponseMessage: "",
```

Comment:
Tiny nit: you can remove zero values where they don't have any meaning to improve readability slightly.

```suggestion
```